R/airtabler.R

```diff
@@ -530,24 +572,29 @@ air_table_funs <- function(base, table_name) {
   res_list
 }
 
-.bind_df <- function(x) {
-  # x = list of data frames
-  if(length(unique(lengths(x))) != 1) {
+.bind_df <- function(dfs) {
+  # dfs = list of data frames
+
+  # add missing columns
+  all_col_names <- unique(unlist(lapply(dfs, names)))
 
-    # add missing columns
-    col_names <- unique(unlist(lapply(x, names)))
-    col_missing <- lapply(x, function(x) setdiff(col_names, names(x)))
+  dfs_m = list()
+  for (i in seq(length(dfs))) {
+    df = dfs[[i]]
 
-    x <- lapply(seq_along(x), function(i) {
-      ret <- x[[i]]
-      for(col in col_missing[[i]]) {
-        ret[[col]] <- NA
+    for (col in all_col_names) {
+      if (class(df[[col]]) == 'data.frame') {
+        row.names(df[[col]]) <- df$id # this takes care of the embedded data.frames
+        next
```

Comment:
Could this just be turned into an `else if` instead of using `next`?

Reply:
This is a very interesting question Austin, made me think about why I chose this approach!

In terms of logic and computation cost, your suggestion makes total sense and the outcome would be equal!

However, I tend to side with the guard pattern as much as I can, mostly because of the benefits of using it in pipelines. Maybe I am abusing it a little...

I tend to process logic more easily this way because I can declare in my head that if a column is of a type `data.frame` that will be dealt with in that branch, and I skip to the next item in the loop.

Simultaneously, I also declare that whatever happens downstream in the loop won't affect the outcome, so I don't need to read it for the `data.frame` case.

I also tend to use `else`, `else if` and `switch` when there are comparisons of the same type because I don't need to think of a different value domain. However, the following if statement in the loop deals with the `value` rather than the `class` of the column. I know that this is a minor distinction...

This thinking comes from `Python` where there isn't a real `switch` structure, though you can do `elif`s.

Guard pattern also tends to flatten the code, and unlinks dependencies, which in this case is not that necessary, we're still using the same object.